raiden/network/proxies/token_networks_registry.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+from binascii import hexlify, unhexlify
+
+import structlog
+
+from raiden.blockchain.abi import (
+    CONTRACT_MANAGER,
+    CONTRACT_TOKEN_NETWORK_REGISTRY,
+    EVENT_TOKEN_ADDED2,
+)
+from raiden.exceptions import (
+    NoTokenManager,
+    TransactionThrew,
+)
+from raiden.utils import (
+    address_decoder,
+    address_encoder,
+    isaddress,
+    pex,
+    privatekey_to_address,
+)
+from raiden.settings import (
+    DEFAULT_POLL_TIMEOUT,
+)
+from raiden.network.proxies.token_network import TokenNetwork
+from raiden.network.rpc.client import check_address_has_code
+from raiden.network.rpc.transactions import (
+    check_transaction_threw,
+    estimate_and_transact,
+)
+from raiden.network.rpc.filters import (
+    new_filter,
+    Filter,
+)
+
+log = structlog.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+class TokenNetworksRegistry:
+    def __init__(
+            self,
+            jsonrpc_client,
+            registry_address,
+            poll_timeout=DEFAULT_POLL_TIMEOUT):
+        # pylint: disable=too-many-arguments
+
+        if not isaddress(registry_address):
+            raise ValueError('registry_address must be a valid address')
+
+        check_address_has_code(jsonrpc_client, registry_address, CONTRACT_TOKEN_NETWORK_REGISTRY)
+
+        proxy = jsonrpc_client.new_contract_proxy(
+            CONTRACT_MANAGER.get_abi(CONTRACT_TOKEN_NETWORK_REGISTRY),
+            address_encoder(registry_address),
+        )
+        CONTRACT_MANAGER.check_contract_version(
+            proxy.call('contract_version').decode(),
+            CONTRACT_TOKEN_NETWORK_REGISTRY
+        )
+
+        self.address = registry_address
+        self.proxy = proxy
+        self.client = jsonrpc_client
+        self.poll_timeout = poll_timeout
+        self.node_address = privatekey_to_address(self.client.privkey)
+
+        self.address_to_channelmanager = dict()
+        self.token_to_channelmanager = dict()
+
+    def manager_address_by_token(self, token_address):
```

Comment:
Shouldn't this (function name and docstring) be edited? This should be `token_to_tokennetwork` or something similar.

Reply:
Same question: should I rename everything, including interface functions now and make integration harder?